src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx

```diff
@@ -84,6 +99,68 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
       });
   };
 
+  // Handlers and Instances have a type attached to the name with '.'
+  // i.e. handler=myhandler.kubernetes
+  validateRuleParams = (parsed: ParsedSearch, rule: IstioRuleDetails): boolean => {
+    if (!parsed.type || !parsed.name || rule.actions.length === 0) {
+      return false;
+    }
+    let validationType = ['handler', 'instance'];
+    if (parsed.type && validationType.indexOf(parsed.type) < 0) {
+      return false;
+    }
+    let splitName = parsed.name.split('.');
+    if (splitName.length !== 2) {
+      return false;
+    }
+    // i.e. handler=myhandler.kubernetes
+    // innerName == myhandler
+    // innerType == kubernetes
+    let innerName = splitName[0];
+    let innerType = splitName[1];
+
+    for (let i = 0; i < rule.actions.length; i++) {
+      if (
+        parsed.type === 'handler' &&
+        rule.actions[i].handler.name === innerName &&
+        rule.actions[i].handler.adapter === innerType
+      ) {
+        return true;
+      }
+      if (parsed.type === 'instance') {
+        for (let j = 0; j < rule.actions[i].instances.length; j++) {
+          if (rule.actions[i].instances[j].name === innerName && rule.actions[i].instances[j].template === innerType) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  };
+
+  // Helper method to extract search urls with format
+  // ?handler=name.handlertype or ?instance=name.instancetype
+  // Those url are expected to be received on this page.
+  parseRuleSearchParams = (): ParsedSearch => {
+    let parsed: ParsedSearch = {};
+    if (this.props.location.search) {
```

Comment:
Just a thought, do you think it would be useful to you the usage of [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) class?

Reply:
Yeah, it makes sense, I will refactor with that.